src/main/java/uk/gov/hmcts/befta/featuretoggle/launchdarkly/LaunchDarklyFeatureToggleService.java

```diff
@@ -2,50 +2,33 @@
 
 import com.launchdarkly.sdk.LDUser;
 import com.launchdarkly.sdk.server.LDClient;
-
-import java.util.List;
-import java.util.stream.Collectors;
-
-import io.cucumber.java.Scenario;
 import uk.gov.hmcts.befta.exception.FeatureToggleCheckFailureException;
-import uk.gov.hmcts.befta.featuretoggle.FeatureToggleInfo;
 import uk.gov.hmcts.befta.featuretoggle.FeatureToggleService;
 
-public class LaunchDarklyFeatureToggleService implements FeatureToggleService {
+public class LaunchDarklyFeatureToggleService implements FeatureToggleService<String, Boolean> {
 
     public static final LaunchDarklyFeatureToggleService INSTANCE = new LaunchDarklyFeatureToggleService();
 
     private static final String BEFTA = "befta";
     private static final String USER = "user";
     private static final String SERVICENAME = "servicename";
 
-    private static final LDUser user = new LDUser.Builder(LaunchDarklyConfig.getEnvironmentName()).firstName(BEFTA)
+    private static final LDUser ldUser = new LDUser.Builder(LaunchDarklyConfig.getEnvironmentName()).firstName(BEFTA)
             .lastName(USER).custom(SERVICENAME, LaunchDarklyConfig.getLDMicroserviceName()).build();
 
-    private static final String LAUNCH_DARKLY_FLAG = "FeatureToggle";
-
     private final LDClient ldClient = LaunchDarklyConfig.getLdInstance();
 
     @Override
-    public FeatureToggleInfo getToggleStatusFor(Scenario scenario) {
-        if (ldClient == null)
-            return null;
-
-        FeatureToggleInfo status = new FeatureToggleInfo();
-        List<String> flagNames = getFeatureFlagsOn(scenario);
-        if (flagNames.isEmpty())
-            return status;
-
-        checkLaunchDarklyConfig(scenario);
-
-        for (String flag : flagNames) {
-            boolean isLDFlagEnabled = ldClient.boolVariation(flag, user, false);
-            status.add(flag, isLDFlagEnabled);
+    public Boolean getToggleStatusFor(String flagId) {
+        if (ldClient == null) {
+            return Boolean.FALSE;
         }
-        return status;
+        checkLaunchDarklyConfig();
+        return ldClient.boolVariation(flagId, ldUser, false);
+
     }
 
-    private void checkLaunchDarklyConfig(Scenario scenario) {
+    private void checkLaunchDarklyConfig() {
```

Comment:
A name like FeatureToggleCheckFailureException and then skipping a scenario doesn't sound consistent. If the LD config is disabled and we want to skip LD-controlled scenarios, there should be a nicer way of communicating this. If the config is incomplete by mistake, we should fail the test suite, not skip the scenario. Perhaps just a naming issue we missed in the original LD work.

Reply:
Hi Mutlu, there might be a couple of functional test case files where we won't be using the Launch Darkly configuration. 
If we go for an early suite failure strategy, then rest of test cases won't be executed.
If we have more than one bad functional test, user might have to fix test cases one by one.

Considering that the functional test cases run at the end of the pipeline, it may significantly increase the time to fix test cases one by one.

Can you please share your thoughts as well.

Reply:
@nitishsingla1234  - I think that is not a sufficient justification to skip scenarios that we are supposed to test but failed to test due to a test suite issue. If we do so, we will be prone to false negatives - builds ending with not test failures but having defects under non-executed scenarios. Moreover, local and preview builds should already be telling of most issues before the work goes to a lengthy master build. So, let's please action this comment to fail the suit if the configuration is not all correct.